locals.tf

```diff
@@ -0,0 +1,3 @@
+locals {
+  create_policy_attachment = (var.users != null || var.roles != null || var.groups != null) ? true : false
+}
```

Comment:
With default values for var.users, var.roles and var.groups being `[ ]` this condition here can be
`(var.users != [] || var.roles != [] || var.groups != []) ? true : false`

Reply:
I've tested this as well but it might not work as is since it will always return true. However grabbing the length of strings works..kindly have a look @patrickmukumbu